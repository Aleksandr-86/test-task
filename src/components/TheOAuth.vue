<script setup lang="ts">
import BaseButton from '@/components/ui/BaseButton.vue'
import { themeIsDark } from '@/store/theme'
</script>

<template>
  <div class="space-y-3">
    <BaseButton label="Войти через Google" icon-src="icons/google.svg" />
    <BaseButton label="Войти через Facebook" icon-src="icons/facebook.svg" />

    <BaseButton
      v-if="themeIsDark"
      label="Войти через Apple"
      icon-src="icons/apple-black.svg" />
    <BaseButton
      v-else
      label="Войти через Apple"
      icon-src="icons/apple-white.svg" />
  </div>
</template>
