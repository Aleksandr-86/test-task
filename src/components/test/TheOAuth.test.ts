import { render } from '@testing-library/vue'
import TheOAuth from '@/components/TheOAuth.vue'

describe('Компонент TheOAuth', () => {
  it('структура соответствует снимку', () => {
    const wrapper = render(TheOAuth, {})

    expect(wrapper).toMatchSnapshot()
  })
})
