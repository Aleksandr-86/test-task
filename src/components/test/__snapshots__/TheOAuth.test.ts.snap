// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Компонент TheOAuth > структура соответствует снимку 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="space-y-3"
      >
        <button
          class="flex h-10 w-full cursor-pointer items-center justify-center rounded-md bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
          type="button"
        >
          <span
            class="mr-2 h-8 w-8"
          >
            <img
              src="/icons/google.svg"
            />
          </span>
           Войти через Google
        </button>
        <button
          class="flex h-10 w-full cursor-pointer items-center justify-center rounded-md bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
          type="button"
        >
          <span
            class="mr-2 h-8 w-8"
          >
            <img
              src="/icons/facebook.svg"
            />
          </span>
           Войти через Facebook
        </button>
        <button
          class="flex h-10 w-full cursor-pointer items-center justify-center rounded-md bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
          type="button"
        >
          <span
            class="mr-2 h-8 w-8"
          >
            <img
              src="/icons/apple-white.svg"
            />
          </span>
           Войти через Apple
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="space-y-3"
    >
      <button
        class="flex h-10 w-full cursor-pointer items-center justify-center rounded-md bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
        type="button"
      >
        <span
          class="mr-2 h-8 w-8"
        >
          <img
            src="/icons/google.svg"
          />
        </span>
         Войти через Google
      </button>
      <button
        class="flex h-10 w-full cursor-pointer items-center justify-center rounded-md bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
        type="button"
      >
        <span
          class="mr-2 h-8 w-8"
        >
          <img
            src="/icons/facebook.svg"
          />
        </span>
         Войти через Facebook
      </button>
      <button
        class="flex h-10 w-full cursor-pointer items-center justify-center rounded-md bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
        type="button"
      >
        <span
          class="mr-2 h-8 w-8"
        >
          <img
            src="/icons/apple-white.svg"
          />
        </span>
         Войти через Apple
      </button>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
